crypto/evp/exchange.c

```diff
@@ -386,3 +386,36 @@ int EVP_PKEY_derive(EVP_PKEY_CTX *ctx, unsigned char *key, size_t *pkeylen)
     M_check_autoarg(ctx, key, pkeylen, EVP_F_EVP_PKEY_DERIVE)
         return ctx->pmeth->derive(ctx, key, pkeylen);
 }
+
+int EVP_KEYEXCH_number(const EVP_KEYEXCH *keyexch)
+{
+    return keyexch->name_id;
+}
+
+int EVP_KEYEXCH_is_a(const EVP_KEYEXCH *keyexch, const char *name)
+{
+    return evp_is_a(keyexch->prov, keyexch->name_id, name);
+}
+
+void EVP_KEYEXCH_do_all_ex(OPENSSL_CTX *libctx,
```

Comment:
Why are we adding an "_ex" function for `EVP_KEYEXCH`?

Reply:
Uniformity.  There's EVP_MD_do_all_ex() and EVP_CIPHER_do_all_ex(), I find it *very* confusing if some of them look like legacy (not ending with `_ex`) and some don't.

Reply:
Maybe rename all the _ex functions to _ctx? But that is also kind of ugly.

Reply:
I also dislike the `_ex` suffix.  However, I dislike a confusing API even more.

So do come up with something better to keep the uniformity.

Reply:
(I'm soooooo tempted to start using the suffix `_v3`)

Reply:
I don't think uniformity is necessary in this case.

Reply:
> Maybe rename all the _ex functions to _ctx? But that is also kind of ugly.

Yes ugly. But I like it better than what is in this PR at the moment.

Reply:
> I don't think uniformity is necessary in this case.

I disagree.  Look, if we have `EVP_EXCHANGE_do_all()`, it looks more like `EVP_CIPHER_do_all()` than `EVP_CIPHER_do_all_ex()`, and `EVP_CIPHER_do_all()` is a very different type of function.  Being a strongly associative mind, that's going to be an anti-pattern I'll come to hate.

But `_ctx` was suggested as an alternative, then `_ctx` it is.  Not sure I find the library context being such a main point, but if that is something we can all agree on, then so be it.

Reply:
Another option, which is more to the point, is `_provided`

Reply:
I saw you, @t8m, so there are now a couple of fixup commits that do the rename